### DISCUSSION TITLE
Linkage issues after migrating recipe to Conan 2

### DISCUSSION THREAD

**Discussion by sobrito** [resiprocate/resiprocate#328] [Q&A] [answered]
Hello,

I'm seeking assistance in migrating the "resiprocate" recipe to Conan 2. I've forked the project and created a specific branch (`conan-recipe-feature`) where I began modifying the CMake to use Conan 2 dependencies, instead of multiple sources like `contrib`, `nuget`, and so on.

In this process, I've set the main package as "resiprocate" and defined dependent packages such as "resip", "rutil", "dum", and "resipares". However, I'm facing linkage problems.

Here are the relevant links:

- Project fork with the specific branch: https://github.com/nerd4ever/resiprocate/tree/conan-recipe-feature
- Updated Conan 2 recipe: https://github.com/nerd4ever/resiprocate/blob/conan-recipe-feature/conanfile.py

I based my implementation on the project's original CMake. Any guidance or suggestions would be greatly appreciated. Thank you!

**Comment by gjasny** [comment]
Hello,
I am traveling right now and do not have access to my Conan 2 recipe for resiprocate. 

I might be able to back to you next week. AFAIR you’ll have to disable the versioned soname cmake option.

Thanks,
Gregor

**Reply by sobrito** [reply]
Hello Gregor,

Thank you for getting in touch. I will be very grateful if you can assist me next week. I am certain that your contribution will be invaluable in solving this issue.

I look forward to your response and safe travels!

Best regards,
Sileno

**Comment by gjasny** [comment]
I merged some more of my conan-enabling patches. Now the following [draft PR](https://github.com/conan-io/conan-center-index/pull/20465) works for me and provides resiprocate as a library (none of the applications). Do you actually need any of the applications? If yes on which operating system?

**Reply by sobrito** [reply]
I don't need any of the applications, just the resiprocate libraries to create other applications, such as a softphone or anything else. Currently, I've managed to do this in the following way:

I cloned the branch I'm using from https://github.com/nerd4ever/resiprocate/tree/1.12.0-1.
I created a recipe in CMake based on the original CMake file. This recipe is available at https://github.com/nerd4ever/conan-center-index/tree/resiprocate/recipes/resiprocate. The CMakeListsConan.cmake file is an adaptation of the original CMake file.
I tested the compilation with Conan on both Windows and Linux (WSL using Ubuntu). However, on Windows, I encountered an issue related to "potp."

**Comment by gjasny** [comment, marked as answer]
I updated my [draft PR](https://github.com/conan-io/conan-center-index/pull/20465) to `cci.20231020` to fix Windows build with c-ares. You need to use c-ares in case you want to also use grpc.
